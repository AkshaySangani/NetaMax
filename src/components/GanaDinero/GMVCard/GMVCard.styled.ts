import styled from "styled-components";

import { Progress } from "@chakra-ui/react";

export const CustomProgressBar = styled(Progress)`
  margin: 20px 64px 5px 45px;
  border-radius: 10px;
  background: linear-gradient(0deg, #412d73 0%, #25094f 100%);
  border: 2px solid #2e6faa;
  div:first-child {
    background: linear-gradient(0deg, #959800 0%, #ffe600 100%);
    height: 65%;
    border-radius: 10px;
    margin: 1.3px 0 0;
  }
`;
