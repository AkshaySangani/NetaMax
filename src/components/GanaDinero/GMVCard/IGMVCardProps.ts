export interface IGMVCardProps {
  /**
   * The card description
   * @type{string}
   */
  cardDescription: string;

  /**
   * The reward amount
   * @type{string}
   **/
  priceAmount: number;

  /**
   * The card's title
   * @type {string}
   */
  title: string;

  /**
   * Sends the user back to home function
   *  @type {() => void}
   * */
  backToHome: () => void;

  /**
   * The progress bar value based on basket total
   * @type {number}
   */
  progressBarValue: number;

  /**
   * If the user complete the challenge or not yet.
   * @type {boolean}
   */
  completed?: boolean;

  /**
   * End date for the challenge
   * @type {Date}
   */
  endChallenge?: Date;

  /**
   * GMV Threshold value to handle the progressbar
   * @type {number}
   */
  GMVThreshold: number;
}
