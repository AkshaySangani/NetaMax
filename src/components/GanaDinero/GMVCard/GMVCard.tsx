import { ReactElement } from "react";

import { LANG_CODE } from "constants/localeFormatConstants";
import IconCoin from "styled/icons/Coin";
import { screenSizes } from "styled/screen";

import { useMediaQuery, Box, Button, Flex, Image, Img, Spacer, Text } from "@chakra-ui/react";

import { CardLayout } from "../CardLayout/CardLayout";
import { BannerText, BannerWrapper, CardWrapper } from "../CardLayout/CardLayout.styled";
import { CustomProgressBar } from "./GMVCard.styled";
import { IGMVCardProps } from "./IGMVCardProps";

/**
 * The GMV card component.
 * @returns {ReactElement} The GMV card component.
 * @param {IGMVCardProps} props The component properties.
 */
export const GMVCard = (props: IGMVCardProps): ReactElement => {
  const {
    title,
    backToHome,
    progressBarValue,
    completed,
    endChallenge,
    cardDescription,
    priceAmount,
    GMVThreshold,
  } = props;
  const [isDesktop] = useMediaQuery(`(min-width: ${screenSizes.mdMin}px)`);

  /**
   * User completed the challenge.
   * @returns {ReactElement} Complete challenge card.
   */
  const challengeCompleted = (): ReactElement => {
    return (
      <CardWrapper>
        <BannerWrapper>
          <BannerText data-testid="card-layout-banner">{title}</BannerText>
        </BannerWrapper>
        <Box paddingTop="20px">
          <Box
            background="#E8E8E8"
            h="220px"
            boxShadow="0px 4px 4px rgba(0, 0, 0, 0.25)"
            borderRadius="12px"
            alignItems="center"
            display="flex"
            justifyContent="center"
            flexDirection="column"
            paddingTop="20px"
          >
            <Image src="/assets/images/icons/clock.png" alt="clock" />
            <Text margin="10px 20px" fontWeight="bold" fontSize="22px">
              Â¡RETO COMPLETADO!
            </Text>
            <Text margin="0px 40px" textAlign="center" fontSize="18px">
              Vuelve maÃ±ana para seguir ganando monetas
            </Text>
          </Box>
        </Box>
      </CardWrapper>
    );
  };

  if (completed) return challengeCompleted();

  return (
    <CardLayout bannerText={title}>
      <Text
        fontSize={isDesktop ? "22px" : "17px"}
        fontWeight="bold"
        color="white"
        mt={["0px", "20px"]}
      >
        {progressBarValue >= GMVThreshold ? "Â¡LLEGASTE A LA META! ðŸŽ‰ " : cardDescription}
      </Text>
      <Text fontSize={isDesktop ? "18px" : "14px"} fontWeight="bold" color="white">
        {progressBarValue >= GMVThreshold
          ? `Finaliza tu pedido actual para ganar $${priceAmount} monetas`
          : ` gana $${priceAmount} monetas para tu prÃ³xima compra`}
      </Text>
      <Box className="progress-bar" pos="relative" w={["100%", "90%"]} mt="20px">
        <CustomProgressBar value={progressBarValue} min={0} max={GMVThreshold} />
        <Img
          src="/assets/images/icono_gana.png"
          height="31px"
          pos="absolute"
          right="30px"
          top="8px"
        />
        {progressBarValue >= GMVThreshold ? (
          <Box pos="absolute" right="41px" top="11px">
            <IconCoin width="25px" height="26px" />
          </Box>
        ) : (
          <Box pos="absolute" right="41px" top="11px">
            <IconCoin width="25px" height="26px" opacity="0.3" />
          </Box>
        )}
      </Box>
      <Flex w="100%" p="0 46px" mt="12px" mb="6px">
        <Box>
          <Text fontSize="15px" fontWeight="bold" color="white">
            Lejos de ganar
          </Text>
        </Box>
        <Spacer />
        <Box>
          <Text fontSize="15px" fontWeight="bold" color="white">
            GANASTE
          </Text>
        </Box>
      </Flex>
      {endChallenge && (
        <Text fontSize={isDesktop ? "18px" : "14px"} fontWeight="bold" m="10px 0" color="white">
          *VÃ¡lido hasta el{" "}
          {endChallenge.toLocaleDateString(LANG_CODE, {
            month: "long",
            day: "numeric",
          })}
        </Text>
      )}
      <Box p="28px 0">
        <Button
          onClick={() => {
            backToHome();
          }}
          borderRadius="16px"
          h="56px"
          w="290px"
          backgroundColor="white"
          color="#3870FF"
          data-testid="gmv-card-button"
        >
          Â¡Ir a comprar!
        </Button>
      </Box>
    </CardLayout>
  );
};
