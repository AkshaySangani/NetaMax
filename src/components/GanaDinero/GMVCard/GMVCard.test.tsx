import { render, cleanup, screen, fireEvent } from "@testing-library/react";
import { GMVCard } from "./GMVCard";
import { IGMVCardProps } from "./IGMVCardProps";

const commonProps: IGMVCardProps = {
  cardDescription: "Recibe 15 en tu primer compra mayor a 200",
  title: "Te regalamos $10",
  progressBarValue: 75,
  GMVThreshold: 200,
  backToHome: jest.fn(),
  priceAmount: 15,
};

const challengeCompletedProps: IGMVCardProps = {
  cardDescription: "Recibe 15 en tu primer compra mayor a 200",
  title: "Te regalamos $10",
  progressBarValue: 200,
  GMVThreshold: 200,
  backToHome: jest.fn(),
  priceAmount: 15,
};

describe("Render GMVCard component", () => {
  test("GMVCard should render properly", () => {
    const { container } = render(<GMVCard {...commonProps} />);
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId("card-layout-banner").textContent).toMatch("Te regalamos $10");
    expect(screen.getByTestId("gmv-card-button").textContent).toMatch("¡Ir a comprar!");
  });

  test("GMVCard should render properly when progress bar value is greather than 200", () => {
    const { container } = render(<GMVCard {...challengeCompletedProps} />);
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId("card-layout-banner").textContent).toMatch("Te regalamos $10");
    expect(screen.getByTestId("gmv-card-button").textContent).toMatch("¡Ir a comprar!");
  });

  afterEach(() => {
    cleanup();
  });
});

describe("Test GMVCard component functions", () => {
  it("executes backToHome when pressing the button", () => {
    render(<GMVCard {...commonProps} />);
    fireEvent.click(screen.getByTestId("gmv-card-button"));
    expect(commonProps.backToHome).toBeCalled();
  });
});
