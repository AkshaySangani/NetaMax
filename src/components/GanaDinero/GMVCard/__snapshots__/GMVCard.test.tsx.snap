// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render GMVCard component GMVCard should render properly 1`] = `
<div>
  <div
    class="sc-bdvvtL bvVYQj css-k008qs"
  >
    <div
      class="sc-gsDKAQ kWXsUD css-0"
    >
      <div
        class="sc-dkPtRN egXpyz css-0"
        data-testid="card-layout-banner"
      >
        Te regalamos $10
      </div>
    </div>
    <div
      class="css-7w6khc"
    >
      <div
        class="css-10klw3m"
      >
        <div
          class="sc-hKwDye dlqxDY css-0"
        >
          <div
            class="sc-eCImPb ftCsbL css-18duxr7"
            data-testid="card-layout-children"
          >
            <p
              class="chakra-text css-r2pa6s"
            >
              Recibe 15 en tu primer compra mayor a 200
            </p>
            <p
              class="chakra-text css-1wjfjnq"
            >
               gana $15 monetas para tu prÃ³xima compra
            </p>
            <div
              class="progress-bar css-1a59n37"
            >
              <div
                class="sc-jRQBWg fVrKET css-kdwx3d"
              >
                <div
                  aria-valuemax="200"
                  aria-valuemin="0"
                  aria-valuenow="75"
                  class="css-1jy0d03"
                  role="progressbar"
                  style="width: 37.5%;"
                />
              </div>
              <img
                class="chakra-image css-1zz5wn"
                src="/assets/images/icono_gana.png"
              />
              <div
                class="css-p2voqj"
              >
                <svg
                  fill="none"
                  height="26px"
                  opacity="0.3"
                  viewBox="0 0 18 18"
                  width="25px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    clip-path="url(#clip0_4082_16816)"
                  >
                    <path
                      d="M16.0139 8.27946C16.0881 9.61603 15.7642 10.9446 15.0833 12.0971C14.4024 13.2496 13.395 14.1743 12.1885 14.7542C10.982 15.3341 9.63064 15.5433 8.3053 15.3552C6.97996 15.1671 5.74016 14.5902 4.74269 13.6975C3.74522 12.8047 3.03489 11.6363 2.70153 10.3398C2.36816 9.04335 2.42674 7.67716 2.86985 6.414C3.31295 5.15084 4.12069 4.04745 5.19091 3.24336C6.26113 2.43928 7.54576 1.97062 8.88234 1.89665C10.6743 1.79787 12.4322 2.41474 13.7695 3.61165C15.1068 4.80855 15.9141 6.48752 16.0139 8.27946Z"
                      fill="#E8AF2E"
                    />
                    <path
                      d="M14.7828 8.34982C14.8433 9.44468 14.5779 10.5329 14.0199 11.4769C13.462 12.4209 12.6367 13.1783 11.6484 13.6532C10.6601 14.1282 9.55308 14.2994 8.46744 14.1453C7.38179 13.9911 6.36624 13.5185 5.54919 12.7872C4.73214 12.0558 4.1503 11.0987 3.87723 10.0367C3.60416 8.97469 3.65213 7.85558 4.01508 6.82085C4.37802 5.78612 5.03965 4.88226 5.91628 4.22354C6.79291 3.56482 7.84518 3.18084 8.94004 3.12016C9.66712 3.07966 10.395 3.18283 11.0822 3.42377C11.7694 3.6647 12.4024 4.03869 12.945 4.52436C13.4876 5.01002 13.9292 5.59784 14.2445 6.25423C14.5598 6.91062 14.7427 7.62272 14.7828 8.34982Z"
                      fill="#F9CA35"
                    />
                    <path
                      d="M10.3709 7.06817C10.3102 6.6389 9.90068 6.44269 9.13916 6.48385C8.37763 6.525 8.00284 6.74693 8.02421 7.15412C8.02712 7.27195 8.06707 7.38586 8.1384 7.47969C8.20973 7.57351 8.3088 7.64248 8.42155 7.6768C8.67269 7.77718 9.17566 7.8787 9.92626 7.98287C10.6769 8.08704 11.2384 8.29964 11.6017 8.6129C11.7847 8.77232 11.932 8.96856 12.034 9.18879C12.136 9.40903 12.1904 9.64831 12.1936 9.89099C12.2158 10.1858 12.1563 10.481 12.0217 10.7443C11.8871 11.0075 11.6825 11.2285 11.4305 11.383C10.8687 11.7586 10.2131 11.9695 9.53775 11.9919C7.79613 12.081 6.7684 11.4652 6.45455 10.1443L7.89147 9.95293C8.03527 10.6484 8.56982 10.9718 9.49514 10.9229C10.36 10.8778 10.7771 10.5617 10.7464 9.97448C10.7462 9.85789 10.7138 9.74361 10.6528 9.64426C10.5918 9.5449 10.5045 9.46431 10.4006 9.41138C10.1847 9.29371 9.68472 9.18011 8.90077 9.07057C8.11912 8.95865 7.54567 8.75661 7.1846 8.46295C7.01155 8.32741 6.87026 8.15564 6.77065 7.95969C6.67103 7.76374 6.61551 7.54836 6.60799 7.32868C6.59666 7.08024 6.64509 6.83271 6.7492 6.60685C6.85331 6.38099 7.01007 6.1834 7.20633 6.03064C7.63087 5.66209 8.26592 5.4561 9.11148 5.41268C10.7832 5.32437 11.6879 5.83178 11.8255 6.93492L10.3709 7.06817Z"
                      fill="#C38728"
                    />
                    <path
                      d="M8.42766 4.33385L8.49219 5.58398L9.86645 5.51305L9.80192 4.26291L8.42766 4.33385Z"
                      fill="#C38728"
                    />
                    <path
                      d="M8.81438 11.7831L8.87891 13.0332L10.2532 12.9623L10.1886 11.7121L8.81438 11.7831Z"
                      fill="#C38728"
                    />
                  </g>
                  <defs>
                    <clippath
                      id="clip0_4082_16816"
                    >
                      <rect
                        fill="white"
                        height="13.539"
                        transform="translate(0 4.77148) rotate(-19.6648)"
                        width="14.1767"
                      />
                    </clippath>
                  </defs>
                </svg>
              </div>
            </div>
            <div
              class="css-1vy2d2r"
            >
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-ofp45a"
                >
                  Lejos de ganar
                </p>
              </div>
              <div
                class="css-17xejub"
              />
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-ofp45a"
                >
                  GANASTE
                </p>
              </div>
            </div>
            <div
              class="css-1vx1p18"
            >
              <button
                class="chakra-button css-p6ec3w"
                data-testid="gmv-card-button"
                type="button"
              >
                Â¡Ir a comprar!
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render GMVCard component GMVCard should render properly when progress bar value is greather than 200 1`] = `
<div>
  <div
    class="sc-bdvvtL bvVYQj css-k008qs"
  >
    <div
      class="sc-gsDKAQ kWXsUD css-0"
    >
      <div
        class="sc-dkPtRN egXpyz css-0"
        data-testid="card-layout-banner"
      >
        Te regalamos $10
      </div>
    </div>
    <div
      class="css-7w6khc"
    >
      <div
        class="css-10klw3m"
      >
        <div
          class="sc-hKwDye dlqxDY css-0"
        >
          <div
            class="sc-eCImPb ftCsbL css-18duxr7"
            data-testid="card-layout-children"
          >
            <p
              class="chakra-text css-r2pa6s"
            >
              Â¡LLEGASTE A LA META! ðŸŽ‰ 
            </p>
            <p
              class="chakra-text css-1wjfjnq"
            >
              Finaliza tu pedido actual para ganar $15 monetas
            </p>
            <div
              class="progress-bar css-1a59n37"
            >
              <div
                class="sc-jRQBWg fVrKET css-kdwx3d"
              >
                <div
                  aria-valuemax="200"
                  aria-valuemin="0"
                  aria-valuenow="200"
                  class="css-1jy0d03"
                  role="progressbar"
                  style="width: 100%;"
                />
              </div>
              <img
                class="chakra-image css-1zz5wn"
                src="/assets/images/icono_gana.png"
              />
              <div
                class="css-p2voqj"
              >
                <svg
                  fill="none"
                  height="26px"
                  viewBox="0 0 18 18"
                  width="25px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    clip-path="url(#clip0_4082_16816)"
                  >
                    <path
                      d="M16.0139 8.27946C16.0881 9.61603 15.7642 10.9446 15.0833 12.0971C14.4024 13.2496 13.395 14.1743 12.1885 14.7542C10.982 15.3341 9.63064 15.5433 8.3053 15.3552C6.97996 15.1671 5.74016 14.5902 4.74269 13.6975C3.74522 12.8047 3.03489 11.6363 2.70153 10.3398C2.36816 9.04335 2.42674 7.67716 2.86985 6.414C3.31295 5.15084 4.12069 4.04745 5.19091 3.24336C6.26113 2.43928 7.54576 1.97062 8.88234 1.89665C10.6743 1.79787 12.4322 2.41474 13.7695 3.61165C15.1068 4.80855 15.9141 6.48752 16.0139 8.27946Z"
                      fill="#E8AF2E"
                    />
                    <path
                      d="M14.7828 8.34982C14.8433 9.44468 14.5779 10.5329 14.0199 11.4769C13.462 12.4209 12.6367 13.1783 11.6484 13.6532C10.6601 14.1282 9.55308 14.2994 8.46744 14.1453C7.38179 13.9911 6.36624 13.5185 5.54919 12.7872C4.73214 12.0558 4.1503 11.0987 3.87723 10.0367C3.60416 8.97469 3.65213 7.85558 4.01508 6.82085C4.37802 5.78612 5.03965 4.88226 5.91628 4.22354C6.79291 3.56482 7.84518 3.18084 8.94004 3.12016C9.66712 3.07966 10.395 3.18283 11.0822 3.42377C11.7694 3.6647 12.4024 4.03869 12.945 4.52436C13.4876 5.01002 13.9292 5.59784 14.2445 6.25423C14.5598 6.91062 14.7427 7.62272 14.7828 8.34982Z"
                      fill="#F9CA35"
                    />
                    <path
                      d="M10.3709 7.06817C10.3102 6.6389 9.90068 6.44269 9.13916 6.48385C8.37763 6.525 8.00284 6.74693 8.02421 7.15412C8.02712 7.27195 8.06707 7.38586 8.1384 7.47969C8.20973 7.57351 8.3088 7.64248 8.42155 7.6768C8.67269 7.77718 9.17566 7.8787 9.92626 7.98287C10.6769 8.08704 11.2384 8.29964 11.6017 8.6129C11.7847 8.77232 11.932 8.96856 12.034 9.18879C12.136 9.40903 12.1904 9.64831 12.1936 9.89099C12.2158 10.1858 12.1563 10.481 12.0217 10.7443C11.8871 11.0075 11.6825 11.2285 11.4305 11.383C10.8687 11.7586 10.2131 11.9695 9.53775 11.9919C7.79613 12.081 6.7684 11.4652 6.45455 10.1443L7.89147 9.95293C8.03527 10.6484 8.56982 10.9718 9.49514 10.9229C10.36 10.8778 10.7771 10.5617 10.7464 9.97448C10.7462 9.85789 10.7138 9.74361 10.6528 9.64426C10.5918 9.5449 10.5045 9.46431 10.4006 9.41138C10.1847 9.29371 9.68472 9.18011 8.90077 9.07057C8.11912 8.95865 7.54567 8.75661 7.1846 8.46295C7.01155 8.32741 6.87026 8.15564 6.77065 7.95969C6.67103 7.76374 6.61551 7.54836 6.60799 7.32868C6.59666 7.08024 6.64509 6.83271 6.7492 6.60685C6.85331 6.38099 7.01007 6.1834 7.20633 6.03064C7.63087 5.66209 8.26592 5.4561 9.11148 5.41268C10.7832 5.32437 11.6879 5.83178 11.8255 6.93492L10.3709 7.06817Z"
                      fill="#C38728"
                    />
                    <path
                      d="M8.42766 4.33385L8.49219 5.58398L9.86645 5.51305L9.80192 4.26291L8.42766 4.33385Z"
                      fill="#C38728"
                    />
                    <path
                      d="M8.81438 11.7831L8.87891 13.0332L10.2532 12.9623L10.1886 11.7121L8.81438 11.7831Z"
                      fill="#C38728"
                    />
                  </g>
                  <defs>
                    <clippath
                      id="clip0_4082_16816"
                    >
                      <rect
                        fill="white"
                        height="13.539"
                        transform="translate(0 4.77148) rotate(-19.6648)"
                        width="14.1767"
                      />
                    </clippath>
                  </defs>
                </svg>
              </div>
            </div>
            <div
              class="css-1vy2d2r"
            >
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-ofp45a"
                >
                  Lejos de ganar
                </p>
              </div>
              <div
                class="css-17xejub"
              />
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-ofp45a"
                >
                  GANASTE
                </p>
              </div>
            </div>
            <div
              class="css-1vx1p18"
            >
              <button
                class="chakra-button css-p6ec3w"
                data-testid="gmv-card-button"
                type="button"
              >
                Â¡Ir a comprar!
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
